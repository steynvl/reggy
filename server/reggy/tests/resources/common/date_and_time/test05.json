{
  "type": "Date and time",
  "information": {
    "dateSeparator": "Hyphen",
    "timeSeparator": "Colon or dot",
    "amPmIndicator": "PM",
    "leadingZeros": "Optional leading zeros",
    "dateFormats": "h:H:n:s:a"
  },
  "generalRegexInfo": {
    "startRegexMatchAt": "Anywhere",
    "endRegexMatchAt": "Anywhere",
    "regexTarget": "Java"
  },
  "generateMethod": "commonUseCases"
}
