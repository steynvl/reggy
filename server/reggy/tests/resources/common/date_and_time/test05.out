Java
Pattern regex = Pattern.compile("(?:1[0-2]|0?[1-9])[:.](?:2[0-3]|[01]?[0-9])[:.](?:[0-5]?[0-9])[:.](?:[0-5]?[0-9])[:.]PM");
(?:1[0-2]|0?[1-9])[:.](?:2[0-3]|[01]?[0-9])[:.](?:[0-5]?[0-9])[:.](?:[0-5]?[0-9])[:.]PM
Perl
my $regex = /(?:1[0-2]|0?[1-9])[:.](?:2[0-3]|[01]?[0-9])[:.](?:[0-5]?[0-9])[:.](?:[0-5]?[0-9])[:.]PM/;
(?:1[0-2]|0?[1-9])[:.](?:2[0-3]|[01]?[0-9])[:.](?:[0-5]?[0-9])[:.](?:[0-5]?[0-9])[:.]PM
POSIX
(?:1[0-2]|0?[1-9])[:.](?:2[0-3]|[01]?[0-9])[:.](?:[0-5]?[0-9])[:.](?:[0-5]?[0-9])[:.]PM
(?:1[0-2]|0?[1-9])[:.](?:2[0-3]|[01]?[0-9])[:.](?:[0-5]?[0-9])[:.](?:[0-5]?[0-9])[:.]PM
Python
regex = re.compile(r'(?:1[0-2]|0?[1-9])[:.](?:2[0-3]|[01]?[0-9])[:.](?:[0-5]?[0-9])[:.](?:[0-5]?[0-9])[:.]PM')
(?:1[0-2]|0?[1-9])[:.](?:2[0-3]|[01]?[0-9])[:.](?:[0-5]?[0-9])[:.](?:[0-5]?[0-9])[:.]PM
JavaScript
const regex = /(?:1[0-2]|0?[1-9])[:.](?:2[0-3]|[01]?[0-9])[:.](?:[0-5]?[0-9])[:.](?:[0-5]?[0-9])[:.]PM/;
(?:1[0-2]|0?[1-9])[:.](?:2[0-3]|[01]?[0-9])[:.](?:[0-5]?[0-9])[:.](?:[0-5]?[0-9])[:.]PM
PHP
$regex = '/(?:1[0-2]|0?[1-9])[:.](?:2[0-3]|[01]?[0-9])[:.](?:[0-5]?[0-9])[:.](?:[0-5]?[0-9])[:.]PM/';
(?:1[0-2]|0?[1-9])[:.](?:2[0-3]|[01]?[0-9])[:.](?:[0-5]?[0-9])[:.](?:[0-5]?[0-9])[:.]PM
Golang
regex, _ := regexp.Compile("(?:1[0-2]|0?[1-9])[:.](?:2[0-3]|[01]?[0-9])[:.](?:[0-5]?[0-9])[:.](?:[0-5]?[0-9])[:.]PM")
(?:1[0-2]|0?[1-9])[:.](?:2[0-3]|[01]?[0-9])[:.](?:[0-5]?[0-9])[:.](?:[0-5]?[0-9])[:.]PM
Rust
let re = Regex::new(r"(?:1[0-2]|0?[1-9])[:.](?:2[0-3]|[01]?[0-9])[:.](?:[0-5]?[0-9])[:.](?:[0-5]?[0-9])[:.]PM").unwrap();
(?:1[0-2]|0?[1-9])[:.](?:2[0-3]|[01]?[0-9])[:.](?:[0-5]?[0-9])[:.](?:[0-5]?[0-9])[:.]PM
C#
var re = new Regex(@"(?:1[0-2]|0?[1-9])[:.](?:2[0-3]|[01]?[0-9])[:.](?:[0-5]?[0-9])[:.](?:[0-5]?[0-9])[:.]PM");
(?:1[0-2]|0?[1-9])[:.](?:2[0-3]|[01]?[0-9])[:.](?:[0-5]?[0-9])[:.](?:[0-5]?[0-9])[:.]PM
Scala
val re = "(?:1[0-2]|0?[1-9])[:.](?:2[0-3]|[01]?[0-9])[:.](?:[0-5]?[0-9])[:.](?:[0-5]?[0-9])[:.]PM".r
(?:1[0-2]|0?[1-9])[:.](?:2[0-3]|[01]?[0-9])[:.](?:[0-5]?[0-9])[:.](?:[0-5]?[0-9])[:.]PM
Kotlin
val regex = Regex("(?:1[0-2]|0?[1-9])[:.](?:2[0-3]|[01]?[0-9])[:.](?:[0-5]?[0-9])[:.](?:[0-5]?[0-9])[:.]PM")
(?:1[0-2]|0?[1-9])[:.](?:2[0-3]|[01]?[0-9])[:.](?:[0-5]?[0-9])[:.](?:[0-5]?[0-9])[:.]PM